brendenwest/lostappleproject
# Mg 18 apple details page

This resolves Issue #18 and #54:

After adding some libraries for Stack Navigation and Gesture Handler, this PR includes modifications to the MainNavigation.js, creates a default Apple Details page, updates the AppleList page to make list items selectable, passes the apple item selected's firebase info to the AppleDetails page, and displays that information. A back button returns the user to the Apple List Page.

_Future updates will enhance the AppleDetails page: Adding the header, including future information that gets added for each apple, and stylings of the elements._

MainNavigation changes:
- In order to include the AppleDetails page with back navigation, the Stack Navigation and Gesture Handler navigation had to be added.
- Main Navigation had to be modified to include the Stack elements to track the AppleDetails page, without adding it to the BottomTabNavigation.

Tested/Developed on:
 - Wind10 laptop, VS Code, Emulator: "Pixel_3_Android_10(AVD) - 10"


To Verify Updates:
- Run` npm install` to install new libraries.
- run `npx react-native start` to launch Metro, and `npx react-native run-android` to launch emulator.
- Navigate to the "Apples" Page in the bottom tab navigation.
- Select an Apple. Touch Opacity is enabled to indicate user selection.
- You will be navigated to the Apple Details Page. All firebase info for the selected apple will be displayed:
- - Apple name
- - Found information (if found, displays year, if not found, displays "???"
- - Location information
- Select the back arrow in the top left corner to return to the Apple List Page.

Not Found:
![image](https://user-images.githubusercontent.com/49133101/140624728-de5eb2c9-c242-4238-abf3-f82522649b07.png)

Found:
![image](https://user-images.githubusercontent.com/49133101/140624722-433a3ee5-9087-4c31-bfe1-b2989aa9f082.png)
